### PR TITLE
Changed component function calls for LiveView compatibility with module and fixed tests

### PR DESCRIPTION
**Pull request by tsenturk**
For test changes, I'm not sure that it's the correct approach or not. Could you please comment about it?

Thank you.

### CODE REVIEW

**Review by theycallmehero — APPROVED** [review]

File: lib/phoenix_components/view.ex [hunk 1]
```diff
@@ -105,17 +105,19 @@ defmodule PhoenixComponents.View do
   def component(namespace, name, attrs, do: block) when is_list(attrs) do
     do_component(namespace, name, block, attrs)
   end
-  
+
   def component(namespace, name, %Phoenix.HTML.Form{} = form, field) when is_atom(field) do
-    do_component(namespace, name, "", [form: form, field: field])
+    do_component(namespace, name, "", form: form, field: field)
```
Comment (ayrat555) [inline lib/phoenix_components/view.ex:110]: no tests 👮

File: lib/phoenix_components/view.ex [hunk 2]
```diff
@@ -105,17 +105,19 @@ defmodule PhoenixComponents.View do
   def component(namespace, name, attrs, do: block) when is_list(attrs) do
```
Comment (ayrat555) [inline lib/phoenix_components/view.ex:105]: no tests for changes 👮

**Review by v0idpwn — APPROVED** [review]
I'd recommend deprecating `component/2`

**Comment by tsenturk** [comment]
When we rename or deprecate `component/2` we may loose these type of usage alternative;
https://github.com/coingaming/phoenix_components/blob/dc1905743a9d30078b581112af105b48282eca79/test/integration/component_test.exs#L9

**Review by hypno2000 — APPROVED** [review]

**Pull request merged**